From ddf0310613457e9a6b42cfb2fbc60588992316bb Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sun, 23 Aug 2015 10:43:37 -0700
Subject: [PATCH 8/9] Fix enum type conversion warnings

/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_util_params_video.c:59:30: error: implicit conversion from enumeration type 'OMX_DISPLAYTRANSFORMTYPE' (aka 'enum OMX_DISPLAYTRANSFORMTYPE') to different enumeration type 'MMAL_DISPLAYTRANSFORM_T' (aka 'enum MMAL_DISPLAYTRANSFORM_T') [-Werror,-Wenum-conversion]
      mmal->transform = omx->transform;
                      ~ ~~~~~^~~~~~~~~
/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_util_params_video.c:63:25: error: implicit conversion from enumeration type 'OMX_DISPLAYMODETYPE' (aka 'enum OMX_DISPLAYMODETYPE') to different enumeration type 'MMAL_DISPLAYMODE_T' (aka 'enum MMAL_DISPLAYMODE_T') [-Werror,-Wenum-conversion]
      mmal->mode = omx->mode;
                 ~ ~~~~~^~~~
/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_util_params_video.c:75:31: error: implicit conversion from enumeration type 'MMAL_DISPLAYTRANSFORM_T' (aka 'enum MMAL_DISPLAYTRANSFORM_T') to different enumeration type 'OMX_DISPLAYTRANSFORMTYPE' (aka 'enum OMX_DISPLAYTRANSFORMTYPE') [-Werror,-Wenum-conversion]
      omx->transform  = mmal->transform;
                      ~ ~~~~~~^~~~~~~~~
/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_util_params_video.c:79:31: error: implicit conversion from enumeration type 'MMAL_DISPLAYMODE_T' (aka 'enum MMAL_DISPLAYMODE_T') to different enumeration type 'OMX_DISPLAYMODETYPE' (aka 'enum OMX_DISPLAYMODETYPE') [-Werror,-Wenum-conversion]
      omx->mode       = mmal->mode;
                      ~ ~~~~~~^~~~

/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/vc/mmal_vc_client.c:102:13: error: implicit conversion from enumeration type
      'VCOS_STATUS_T' to different enumeration type 'MMAL_STATUS_T' [-Werror,-Wenum-conversion]
   status = vcos_semaphore_create(&waitpool->sem, VCOS_FUNCTION,
          ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/vc/mmal_vc_client.c:110:16: error: implicit conversion from enumeration type
      'VCOS_STATUS_T' to different enumeration type 'MMAL_STATUS_T' [-Werror,-Wenum-conversion]
      status = vcos_semaphore_create(&waitpool->waiters[i].sem,
             ~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_core.c:284:17: error: implicit conversion from enumeration
      type 'MMAL_STATUS_T' to different enumeration type 'OMX_ERRORTYPE' (aka 'enum OMX_ERRORTYPE') [-Werror,-Wenum-conversion]
         return mmalomx_get_port_settings(port, param);
         ~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_core.c:478:17: error: implicit conversion from enumeration
      type 'MMAL_STATUS_T' to different enumeration type 'OMX_ERRORTYPE' (aka 'enum OMX_ERRORTYPE') [-Werror,-Wenum-conversion]
         return mmalomx_set_port_settings(port, param);
         ~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_parameters.c:173:14: error: implicit conversion from
      enumeration type 'MMAL_STATUS_T' to different enumeration type 'OMX_ERRORTYPE' (aka 'enum OMX_ERRORTYPE') [-Werror,-Wenum-conversion]
      return xlat->fn.custom(MMALOMX_PARAM_MAPPING_TO_OMX, xlat, mmal_header,
      ~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_parameters.c:557:17: error: implicit conversion from
      enumeration type 'MMAL_STATUS_T' to different enumeration type 'OMX_ERRORTYPE' (aka 'enum OMX_ERRORTYPE') [-Werror,-Wenum-conversion]
         return mmal_port_format_commit(port->mmal);
         ~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/home/kraj/work/angstrom/build/tmp-angstrom-glibc/work/raspberrypi2-angstrom-linux-gnueabi/userland/git-r5/git/interface/mmal/openmaxil/mmalomx_parameters.c:564:17: error: implicit conversion from
      enumeration type 'MMAL_STATUS_T' to different enumeration type 'OMX_ERRORTYPE' (aka 'enum OMX_ERRORTYPE') [-Werror,-Wenum-conversion]
         return mmal_port_format_commit(port->mmal);
         ~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 interface/mmal/openmaxil/mmalomx_core.c              | 4 ++--
 interface/mmal/openmaxil/mmalomx_parameters.c        | 6 +++---
 interface/mmal/openmaxil/mmalomx_util_params_video.c | 8 ++++----
 interface/mmal/vc/mmal_vc_client.c                   | 4 ++--
 4 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/interface/mmal/openmaxil/mmalomx_core.c b/interface/mmal/openmaxil/mmalomx_core.c
index da66b0b..3a72a2e 100644
--- a/interface/mmal/openmaxil/mmalomx_core.c
+++ b/interface/mmal/openmaxil/mmalomx_core.c
@@ -281,7 +281,7 @@ static OMX_ERRORTYPE mmalomx_ComponentGetParameter(
       {
          OMX_PARAM_PORTDEFINITIONTYPE *param = (OMX_PARAM_PORTDEFINITIONTYPE *)pParam;
          PARAM_GET_PORT(port, component, param->nPortIndex);
-         return mmalomx_get_port_settings(port, param);
+         return (OMX_ERRORTYPE)mmalomx_get_port_settings(port, param);
       }
       return OMX_ErrorNone;
       break;
@@ -475,7 +475,7 @@ static OMX_ERRORTYPE mmalomx_ComponentSetParameter(
       {
          OMX_PARAM_PORTDEFINITIONTYPE *param = (OMX_PARAM_PORTDEFINITIONTYPE *)pParam;
          PARAM_GET_PORT(port, component, param->nPortIndex);
-         return mmalomx_set_port_settings(port, param);
+         return (OMX_ERRORTYPE)mmalomx_set_port_settings(port, param);
       }
       return OMX_ErrorNone;
       break;
diff --git a/interface/mmal/openmaxil/mmalomx_parameters.c b/interface/mmal/openmaxil/mmalomx_parameters.c
index a91b68c..f0bd17f 100644
--- a/interface/mmal/openmaxil/mmalomx_parameters.c
+++ b/interface/mmal/openmaxil/mmalomx_parameters.c
@@ -170,7 +170,7 @@ static OMX_ERRORTYPE mmalomx_parameter_get_xlat(MMALOMX_COMPONENT_T *component,
 
    if (xlat->fn.custom)
    {
-      return xlat->fn.custom(MMALOMX_PARAM_MAPPING_TO_OMX, xlat, mmal_header,
+      return (OMX_ERRORTYPE)xlat->fn.custom(MMALOMX_PARAM_MAPPING_TO_OMX, xlat, mmal_header,
          pParam, mmal_port);
    }
 
@@ -554,14 +554,14 @@ OMX_ERRORTYPE mmalomx_parameter_set(MMALOMX_COMPONENT_T *component,
          port->mmal->format->es->video.par.num = param->nX;
          port->mmal->format->es->video.par.den = param->nY;
          mmal_rational_simplify(&port->mmal->format->es->video.par);
-         return mmal_port_format_commit(port->mmal);
+         return (OMX_ERRORTYPE)mmal_port_format_commit(port->mmal);
       }
    case OMX_IndexParamColorSpace:
       {
          OMX_PARAM_COLORSPACETYPE *param = (OMX_PARAM_COLORSPACETYPE *)pParam;
          PARAM_GET_PORT(port, component, param->nPortIndex);
          port->mmal->format->es->video.color_space = mmalil_omx_color_space_to_mmal(param->eColorSpace);
-         return mmal_port_format_commit(port->mmal);
+         return (OMX_ERRORTYPE)mmal_port_format_commit(port->mmal);
       }
    case OMX_IndexParamBrcmVideoCroppingDisable:
       {
diff --git a/interface/mmal/openmaxil/mmalomx_util_params_video.c b/interface/mmal/openmaxil/mmalomx_util_params_video.c
index f088296..83e3724 100644
--- a/interface/mmal/openmaxil/mmalomx_util_params_video.c
+++ b/interface/mmal/openmaxil/mmalomx_util_params_video.c
@@ -56,11 +56,11 @@ static MMAL_STATUS_T mmalomx_param_mapping_displayregion(MMALOMX_PARAM_MAPPING_D
       mmal->set = omx->set;
       mmal->display_num = omx->num;
       mmal->fullscreen = omx->fullscreen;
-      mmal->transform = omx->transform;
+      mmal->transform = (MMAL_DISPLAYTRANSFORM_T)omx->transform;
       rect_to_mmal(&mmal->dest_rect, &omx->dest_rect);
       rect_to_mmal(&mmal->src_rect, &omx->src_rect);
       mmal->noaspect = omx->noaspect;
-      mmal->mode = omx->mode;
+      mmal->mode = (MMAL_DISPLAYMODE_T)omx->mode;
       mmal->pixel_x = omx->pixel_x;
       mmal->pixel_y = omx->pixel_y;
       mmal->layer = omx->layer;
@@ -72,11 +72,11 @@ static MMAL_STATUS_T mmalomx_param_mapping_displayregion(MMALOMX_PARAM_MAPPING_D
       omx->set        = mmal->set;
       omx->num        = mmal->display_num;
       omx->fullscreen = mmal->fullscreen;
-      omx->transform  = mmal->transform;
+      omx->transform  = (OMX_DISPLAYTRANSFORMTYPE)mmal->transform;
       rect_to_omx(&omx->dest_rect, &mmal->dest_rect);
       rect_to_omx(&omx->src_rect, &mmal->src_rect);
       omx->noaspect   = mmal->noaspect;
-      omx->mode       = mmal->mode;
+      omx->mode       = (OMX_DISPLAYMODETYPE)mmal->mode;
       omx->pixel_x    = mmal->pixel_x;
       omx->pixel_y    = mmal->pixel_y;
       omx->layer      = mmal->layer;
diff --git a/interface/mmal/vc/mmal_vc_client.c b/interface/mmal/vc/mmal_vc_client.c
index b60544b..bd27fc5 100644
--- a/interface/mmal/vc/mmal_vc_client.c
+++ b/interface/mmal/vc/mmal_vc_client.c
@@ -99,7 +99,7 @@ static MMAL_STATUS_T create_waitpool(MMAL_WAITPOOL_T *waitpool)
    MMAL_STATUS_T status;
    int i;
 
-   status = vcos_semaphore_create(&waitpool->sem, VCOS_FUNCTION,
+   status = (MMAL_STATUS_T)vcos_semaphore_create(&waitpool->sem, VCOS_FUNCTION,
                                   MAX_WAITERS);
    if (status != MMAL_SUCCESS)
       return status;
@@ -107,7 +107,7 @@ static MMAL_STATUS_T create_waitpool(MMAL_WAITPOOL_T *waitpool)
    for (i=0; i<MAX_WAITERS; i++)
    {
       waitpool->waiters[i].inuse = 0;
-      status = vcos_semaphore_create(&waitpool->waiters[i].sem,
+      status = (MMAL_STATUS_T)vcos_semaphore_create(&waitpool->waiters[i].sem,
                                      "mmal waiter", 0);
       if (status != MMAL_SUCCESS)
          break;
-- 
2.1.4

